daemon/vpn/nordlynx/libtelio/cgo.go

```diff
@@ -1,11 +1,9 @@
-//go:build !internal
-
 package libtelio
 
-// #cgo amd64 LDFLAGS: -L${SRCDIR}/../../../../bin/deps/foss/amd64/latest -lfoss
-// #cgo 386 LDFLAGS: -L${SRCDIR}/../../../../bin/deps/foss/i386/latest -lfoss
-// #cgo arm LDFLAGS: -L${SRCDIR}/../../../../bin/deps/foss/armel/latest -lfoss
-// #cgo arm LDFLAGS: -L${SRCDIR}/../../../../bin/deps/foss/armhf/latest -lfoss
-// #cgo arm64 LDFLAGS: -L${SRCDIR}/../../../../bin/deps/foss/aarch64/latest -lfoss
+// #cgo amd64 LDFLAGS: -L${SRCDIR}/../../../../bin/deps/lib/amd64/latest -ltelio -lsqlite3
```

Comment:
Does telio use sqlite directly?

Reply:
I also simplified the cgo directives since we don't have paths, we don't need separate directives per architecture. Appropriate path for .so files is set [here](https://github.com/NordSecurity/nordvpn-linux/pull/516/files#diff-83cdd3dee86dfeb38d2c6a19456ae7c80006b3102c94116b2b5893c16df0b772R93) for build and [here](https://github.com/NordSecurity/nordvpn-linux/pull/516/files#diff-ec5c70eec5eb9ea4cfca52a91194292392ffee01f5f743194ebd6d3c64bb2706R35-R38) for tests

Reply:
Reviewing this again. It seems that sqlite is nowhere used in [libtelio](https://github.com/NordSecurity/libtelio/blob/main/Cargo.lock). For public version it should not be needed

Reply:
I'll fix the conflicts and take a look again, but as far as I remember I was getting linking issues without. Will double check after fixing conflicts

Reply:
Yes, it's failing without specifying `sqlite3`. You can take a look on the latest .so artifact of libtelio

When I do `nm -D ./libtelio.so | rg sqlite3`, I'm getting the following output:
```
         U sqlite3_bind_blob
         U sqlite3_bind_double
         U sqlite3_bind_int64
         U sqlite3_bind_null
         U sqlite3_bind_parameter_count
         U sqlite3_bind_text
         U sqlite3_bind_zeroblob
         U sqlite3_busy_timeout
         U sqlite3_changes
         U sqlite3_close
         U sqlite3_column_blob
         U sqlite3_column_bytes
         U sqlite3_column_count
         U sqlite3_column_double
         U sqlite3_column_int64
         U sqlite3_column_name
         U sqlite3_column_text
         U sqlite3_column_type
         U sqlite3_config
         U sqlite3_errmsg
         U sqlite3_extended_result_codes
         U sqlite3_finalize
         U sqlite3_get_autocommit
         U sqlite3_initialize
         U sqlite3_libversion_number
         U sqlite3_mutex_alloc
         U sqlite3_mutex_free
         U sqlite3_open_v2
         U sqlite3_prepare_v2
         U sqlite3_profile
         U sqlite3_reset
         U sqlite3_step
         U sqlite3_threadsafe
```

and those are the missing symbols in the linker error:
```
/usr/local/go/pkg/tool/linux_amd64/link: running x86_64-linux-gnu-gcc failed: exit status 1
/usr/bin/ld: warning: libsqlite3.so, needed by /opt/daemon/vpn/nordlynx/libtelio/../../../../bin/deps/lib/amd64/latest/libtelio.so, not found (try using -rpath or -rpath-link)
/usr/bin/ld: /opt/daemon/vpn/nordlynx/libtelio/../../../../bin/deps/lib/amd64/latest/libtelio.so: undefined reference to `sqlite3_column_blob'
/usr/bin/ld: /opt/daemon/vpn/nordlynx/libtelio/../../../../bin/deps/lib/amd64/latest/libtelio.so: undefined reference to `sqlite3_column_int64'
/usr/bin/ld: /opt/daemon/vpn/nordlynx/libtelio/../../../../bin/deps/lib/amd64/latest/libtelio.so: undefined reference to `sqlite3_bind_double'
/usr/bin/ld: /opt/daemon/vpn/nordlynx/libtelio/../../../../bin/deps/lib/amd64/latest/libtelio.so: undefined reference to `sqlite3_bind_blob'
/usr/bin/ld: /opt/daemon/vpn/nordlynx/libtelio/../../../../bin/deps/lib/amd64/latest/libtelio.so: undefined reference to `sqlite3_column_type'
/usr/bin/ld: /opt/daemon/vpn/nordlynx/libtelio/../../../../bin/deps/lib/amd64/latest/libtelio.so: undefined reference to `sqlite3_reset'
/usr/bin/ld: /opt/daemon/vpn/nordlynx/libtelio/../../../../bin/deps/lib/amd64/latest/libtelio.so: undefined reference to `sqlite3_bind_null'
/usr/bin/ld: /opt/daemon/vpn/nordlynx/libtelio/../../../../bin/deps/lib/amd64/latest/libtelio.so: undefined reference to `sqlite3_close'
/usr/bin/ld: /opt/daemon/vpn/nordlynx/libtelio/../../../../bin/deps/lib/amd64/latest/libtelio.so: undefined reference to `sqlite3_column_name'
/usr/bin/ld: /opt/daemon/vpn/nordlynx/libtelio/../../../../bin/deps/lib/amd64/latest/libtelio.so: undefined reference to `sqlite3_bind_parameter_count'
/usr/bin/ld: /opt/daemon/vpn/nordlynx/libtelio/../../../../bin/deps/lib/amd64/latest/libtelio.so: undefined reference to `sqlite3_extended_result_codes'
/usr/bin/ld: /opt/daemon/vpn/nordlynx/libtelio/../../../../bin/deps/lib/amd64/latest/libtelio.so: undefined reference to `sqlite3_libversion_number'
/usr/bin/ld: /opt/daemon/vpn/nordlynx/libtelio/../../../../bin/deps/lib/amd64/latest/libtelio.so: undefined reference to `sqlite3_column_double'
/usr/bin/ld: /opt/daemon/vpn/nordlynx/libtelio/../../../../bin/deps/lib/amd64/latest/libtelio.so: undefined reference to `sqlite3_column_text'
/usr/bin/ld: /opt/daemon/vpn/nordlynx/libtelio/../../../../bin/deps/lib/amd64/latest/libtelio.so: undefined reference to `sqlite3_bind_text'
/usr/bin/ld: /opt/daemon/vpn/nordlynx/libtelio/../../../../bin/deps/lib/amd64/latest/libtelio.so: undefined reference to `sqlite3_profile'
/usr/bin/ld: /opt/daemon/vpn/nordlynx/libtelio/../../../../bin/deps/lib/amd64/latest/libtelio.so: undefined reference to `sqlite3_prepare_v2'
/usr/bin/ld: /opt/daemon/vpn/nordlynx/libtelio/../../../../bin/deps/lib/amd64/latest/libtelio.so: undefined reference to `sqlite3_errmsg'
/usr/bin/ld: /opt/daemon/vpn/nordlynx/libtelio/../../../../bin/deps/lib/amd64/latest/libtelio.so: undefined reference to `sqlite3_bind_int64'
/usr/bin/ld: /opt/daemon/vpn/nordlynx/libtelio/../../../../bin/deps/lib/amd64/latest/libtelio.so: undefined reference to `sqlite3_busy_timeout'
/usr/bin/ld: /opt/daemon/vpn/nordlynx/libtelio/../../../../bin/deps/lib/amd64/latest/libtelio.so: undefined reference to `sqlite3_changes'
/usr/bin/ld: /opt/daemon/vpn/nordlynx/libtelio/../../../../bin/deps/lib/amd64/latest/libtelio.so: undefined reference to `sqlite3_mutex_free'
/usr/bin/ld: /opt/daemon/vpn/nordlynx/libtelio/../../../../bin/deps/lib/amd64/latest/libtelio.so: undefined reference to `sqlite3_mutex_alloc'
/usr/bin/ld: /opt/daemon/vpn/nordlynx/libtelio/../../../../bin/deps/lib/amd64/latest/libtelio.so: undefined reference to `sqlite3_column_bytes'
/usr/bin/ld: /opt/daemon/vpn/nordlynx/libtelio/../../../../bin/deps/lib/amd64/latest/libtelio.so: undefined reference to `sqlite3_open_v2'
/usr/bin/ld: /opt/daemon/vpn/nordlynx/libtelio/../../../../bin/deps/lib/amd64/latest/libtelio.so: undefined reference to `sqlite3_get_autocommit'
/usr/bin/ld: /opt/daemon/vpn/nordlynx/libtelio/../../../../bin/deps/lib/amd64/latest/libtelio.so: undefined reference to `sqlite3_step'
/usr/bin/ld: /opt/daemon/vpn/nordlynx/libtelio/../../../../bin/deps/lib/amd64/latest/libtelio.so: undefined reference to `sqlite3_finalize'
/usr/bin/ld: /opt/daemon/vpn/nordlynx/libtelio/../../../../bin/deps/lib/amd64/latest/libtelio.so: undefined reference to `sqlite3_config'
/usr/bin/ld: /opt/daemon/vpn/nordlynx/libtelio/../../../../bin/deps/lib/amd64/latest/libtelio.so: undefined reference to `sqlite3_column_count'
/usr/bin/ld: /opt/daemon/vpn/nordlynx/libtelio/../../../../bin/deps/lib/amd64/latest/libtelio.so: undefined reference to `sqlite3_bind_zeroblob'
/usr/bin/ld: /opt/daemon/vpn/nordlynx/libtelio/../../../../bin/deps/lib/amd64/latest/libtelio.so: undefined reference to `sqlite3_initialize'
/usr/bin/ld: /opt/daemon/vpn/nordlynx/libtelio/../../../../bin/deps/lib/amd64/latest/libtelio.so: undefined reference to `sqlite3_threadsafe'
collect2: error: ld returned 1 exit status

Error: exit code 1
```

Reply:
Added `cgo_internal.go` which includes the sqlite, public  build does not need it, so regular `cgo.go` does not include it now